docs/template.html

```diff
@@ -477,8 +481,57 @@
         toggleClass(menuLink, active);
       };
 
+      /**
+       * search
+       */
+
+      // via: https://css-tricks.com/snippets/javascript/get-url-variables/
+      function getQueryVariable(variable)
+      {
+         var query = window.location.search.substring(1);
+         var vars = query.split("&");
+         for (var i=0;i<vars.length;i++) {
+           var pair = vars[i].split("=");
+           if(pair[0] == variable){return pair[1];}
+         }
+         return(false);
+      }
+
+      function htmlToElement(html) {
+          var template = document.createElement('template');
+          html = html.trim(); // Never return a text node of whitespace as the result
+          template.innerHTML = html;
+          return template.content.firstChild;
+      }
+
+      const displayResult = function(result){
+        document.getElementById('searchResults').innerHtml = '';
+        result.map(function(el){
+          var li = htmlToElement('<li><a href="'+el.doc.url+'">'+el.doc.url+'</a></li>')
+          document.getElementById('searchResults').appendChild(li);
+        });
+      }
+
+      var searchIndex = elasticlunr(function () {
+        this.addField('content');
+        this.addField('url');
+        this.setRef('id');
+      });
+
+      siteData.map(function(el){
+        searchIndex.addDoc(el);
+      });
+
+      var searchTerm = getQueryVariable('search').replace(/\+/g, " ");
+      searchTerm = decodeURIComponent(searchTerm);
+      if(searchTerm){
+        document.getElementById('searchTerm').value = searchTerm;
+        var result = searchIndex.search(searchTerm);
+        displayResult(result);
+      }
+
```

Comment:
Same for all these, as mentioned before, this specific for a template that has search with the search index embedded in the HTML, I'd appreciate if you published a separate template that we could link to from the README that features this search and keep this core just simple.

Reply:
Great - thanks for the feedback. I'll work on this and resubmit.